Initial translation support

joaorb64 opened the pull request:
I was finally able to understand how translations work in Qt!

I've created 2 scripts under `src/i18n/`:
- `gen_translation_files.py` generates `ts` files, which should be opened using `Qt 5 Linguistic`. It detects translatable text in all `ui` files and `py` files (all strings should be encapsulated in a specific function call)
- `gen_translation_binaries.py` generates `qm` files from the `ts` files. These are binary files that handle the translation and are the ones we can import in the program itself

These scripts should be executed from the project's root directory. They use `lupdate` and `lrelease` commands. You'll have to see how to install these in your system. Probably included in the Qt suite.

Checklist:

- [x] Encapsulate strings for translation 
- [x] Create translations at least for pt-BR and fr (since I suppose @Wolfy76700 and @Kekwel might be around for this :eyes: )
- [x] Create a class to handle translations: both auto-detect system language and force a language
- [x] Add setting to select language (maybe have a setting for the UI and other for the assets. Example: you want to use the program in French for Smash Ultimate but still have Pokémon names in English in the program output)
- [x] Load country name translations (?) (the countries json has locale settings)
- [x] Load assets translations

joaorb64 commented:
![image](https://user-images.githubusercontent.com/7636440/167727913-9981431b-2cfd-44ee-bd40-34055c36bbb8.png)

joaorb64 commented:
Merged changes from `main`

joaorb64 commented:
Translation to German also added!

Wolfy76700 commented:
Note to translators: If you can please translate the rounds lists present in `/assets/locale`, I keep forgetting about it myself ^^

joaorb64 commented:
Added language selection menus, which handle saving/loading settings. Implemented a remapping system, adding all Spanish languages from the Americas to a group "es_LA", which is used in the SSBU assets pack (es_LA != es_ES).

Program language can be different from export language, and for any names different from their English counterparts the program will show both. Which means that typing to search for a character or country will work for either language!

Now only needs to show a message saying you need to restart, and maybe see if we can implement a restart too.

joaorb64 commented:
Adding a note here that @Wolfy76700 reported it crashing on startup on Windows, and my guess is that we need to add utf-8 flags when loading the JSON files in TSHLocaleHelper. I don't know when I'll be able to do this so adding this reminder here :P

Wolfy76700 commented:
NB: There are a couple of untranslated strings added with the last PR which need to be cleaned up

Additionally, [round name text](https://github.com/joaorb64/TournamentStreamHelper/tree/feature/ui-translations/assets/locale/fr) still need to be translated to work with the feature (At least in Japanese and French where the round name text is translated in the preview ^^)

joaorb64 commented:
Let's merge this and work on more translations long-term/as people request/contribute! I believe the base is solid enough!

Wolfy76700 commented:
Side note: Should we add a section in the Wiki on how to contribute to translations?

joaorb64 commented:
Yeah, sounds like a good idea!
